Update drift guide for nlp (#2595)
deepchecks/deepchecks: docs/source/general/guides/drift_guide.rst

* `Which Types of Drift Are There? <#which-types-of-drift-are-there>`__
* `How Do You Detect Drift? <#how-do-you-detect-drift>`__
* `How Can I Use Deepchecks to Detect Drift? <#how-can-i-use-deepchecks-to-detect-drift>`__

>     * `Tabular Data <#tabular-data>`__

Reviewer: bullets should be of a different kind (and I think that the indentation has to be of exactly by 2)

Author: But you were right about double space only. Both work but this looks better

All of these checks appear also in the `deepchecks interactive demo <https://checks-demo.deepchecks.com>`__, where you can
insert corruption into the data and see the checks at work.

Text (NLP) Data
---------------

> Regarding `data <#data-drift>`__ or `concept drift <#concept-drift>`__

Reviewer: did you build docs and check if all links work?
+
Can you add a pdf of this page? so it can be easily reviewed within the context

Reviewer: submitting review, will continue it on pdf

Author: Added PDF.
About links - checked locally and made sure those that don't work locally (ref) are phrased as in other places. Will make sure with makedocs before merging.

> In text data, we can't measure drift on the text directly, as text is not structured data that can be measured.

Reviewer: "that can be measured":
suggesting in replacing this with:
that measures/metrics can directly be calculated on/for.
or
and thus doesn't have inherent/straight-forward ways to calculate these measures.

Author: changed to "that can be easily quantified or compared", which seemed more accurate and less repetitive IMO

However, we can use different methods to represent the text as a structured variable, and then measure drift on that variable.

> In deepchecks-NLP, we use 2 such methods: :ref:`Text Embeddings <nlp__embeddings_guide>` and :ref:`Text Properties <nlp__properties_guide>`.

Reviewer: ```suggestion
In deepchecks-nlp, we use 2 such methods: 

- :ref:`Text Embeddings <nlp__embeddings_guide>`
- :ref:`Text Properties <nlp__properties_guide>`.
```

Author: I think I'm against it - as there are similar bullets after the next paragraph, may seem unnecessarily long

Both methods have their pros and cons when used to measure drift: Properties are more explainable, but will not necessarily
capture all the information in the text. Embeddings are able to find more complex patterns in the text, but these
patterns may be difficult to explain. Therefore, we recommend to use both methods to detect
`data <#data-drift>`__ or `concept drift <#concept-drift>`__:

#. The :ref:`Text Embeddings Drift Check <nlp__embeddings_drift>` uses embeddings to measure drift using a
   `domain classifier <#detection-by-domain-classifier>`__
#. The :ref:`Text Properties Drift Check <nlp__properties_drift>` uses properties to measure drift using
   `univariate measures <#detection-by-univariate-measure>`__

For drift in your label's distribution, deepchecks offers the :ref:`nlp__label_drift`,
which uses `univariate measures <#detection-by-univariate-measure>`__.

> In cases where the label is not available, we strongly recommend to also use the :ref:`nlp__prediction_drift`,

Reviewer: ```suggestion
In cases where the labels are not available, we strongly recommend to also use the :ref:`nlp__prediction_drift`,
```

    result = check.run(train_dataset=train_dataset, test_dataset=test_dataset, model=model)

Text (NLP) Checks
-----------------

:ref:`nlp__embeddings_drift`:

.. code-block:: python

    # Load the embeddings from a file:
    train_dataset.set_embeddings('my_train_embeddings_file.npy')
    test_dataset.set_embeddings('my_test_embeddings_file.npy')

    # If you do not have a model to extract embeddings from, you can use the deepchecks default embeddings:
    train_dataset.calculate_default_embeddings()
    test_dataset.calculate_default_embeddings()

>     # For more on loading embeddings, see the :ref:`nlp__embeddings_guide'.

Reviewer: ```suggestion
    # For more on loading embeddings, see the :ref:`nlp__embeddings_guide`.
```

>     # Calculate the check:

Reviewer: ```suggestion
    # Run the check:
```
